composites/OnboardingWizard/helpers/postJSON.js

```diff
@@ -46,46 +41,60 @@ let postJSONFetch = ( url, data ) => {
 		}
 	);
 };
+
 /**
  * Handles JSON request the jQuery way.
  *
  * @param {string} url  The endpoint to send the data to.
+ * @param {object} headers Object containing the headers for the request.
  * @param {Object} data The JSON object to send to the server.
  *
  * @returns {Promise} A Promise, if the request is successful the promise is resolved, else it's rejected.
  */
-let postJSONjQuery = ( url, data ) => {
-	let promise = new Promise( ( resolve, reject )=> {
-		jQuery.post( { url, dataType: "json", data } )
-		      .done(
-			      ( response ) => {
-				      resolve( response );
-			      }
-		      )
-		      .fail(
-			      () => {
-				      reject( "Wrong request" );
-			      }
-		      );
+let postJSONjQuery = ( url, headers, data ) => {
+	return new Promise( ( resolve, reject )=> {
+		jQuery.ajax( {
+			method: "PUT",
+			url,
+			dataType: "json",
+			contentType : 'application/json',
+			beforeSend: function ( xhr ) {
+				jQuery.each( headers, (headerName, headerValue) => {
+					xhr.setRequestHeader(headerName, headerValue);
+				});
+			},
+			data
+		} )
+	      .done(
+		      ( response ) => {
+			      resolve( response );
+		      }
+	      )
+	      .fail(
+		      () => {
+			      reject( "Wrong request" );
+		      }
+	      );
 	} );
-	return promise;
 };
+
 /**
  * Wrapper method when fetch should be used.
  *
  * @param {string} url  The endpoint to send the data to.
+ * @param {object} headers Object containing the headers for the request.
  * @param {Object} data The JSON object to send to the server.
  *
  * @returns {Promise} Returns a wrapped promise.
  */
-let postJSON = ( url, data = {} ) => {
+let postJSON = ( url, headers = {}, data = {} ) => {
```

Comment:
I personally would put headers as the third parameter, since you will most likely pass your own data to a POST/PUT request, but are less likely to add extra headers. This prevents the user from having to add an unnecessary parameter.